users/src/main/java/com/familywarehouse/users/controller/UserController.java

```diff
@@ -0,0 +1,150 @@
+package com.familywarehouse.users.controller;
+
+import com.familywarehouse.users.constants.UserConstants;
+import com.familywarehouse.users.dto.ErrorResponseDto;
+import com.familywarehouse.users.dto.ResponseDto;
+import com.familywarehouse.users.dto.UserDto;
+import com.familywarehouse.users.service.UserService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.Min;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.data.domain.Page;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@Slf4j
+@RestController
+@RequestMapping(value = "/users", produces = {MediaType.APPLICATION_JSON_VALUE})
+@AllArgsConstructor
+@Validated
+public class UserController {
+
+    private final UserService userService;
+
+    @Operation(
+            summary = "Create an user in Family-Warehouse",
+            description = "REST API to create User"
+    )
+    @ApiResponse(
+            responseCode = "201",
+            description = "HTTP Status CREATED"
+    )
+    @PostMapping
+    public ResponseEntity<ResponseDto> createUser(@Valid @RequestBody UserDto userDto) {
```

Comment:
you don't need to have ResponseEntity since we use @RestController. Annotation approach is preferable if you don't have dynamic responce headers